dora/core/server/worker/src/main/java/alluxio/worker/dora/DoraMetaManager.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.dora;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.PageId;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.file.FileId;
+import alluxio.proto.meta.DoraMeta;
+import alluxio.proto.meta.DoraMeta.FileStatus;
+import alluxio.underfs.Fingerprint;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystem;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * The Dora metadata manager that orchestrates the metadata operations and provides the
+ * concurrency control.
```

Comment:
I assume concurrency control is not provided at this moment?

Reply:
yeah i removed the false comment